sdcm/cluster.py

```diff
@@ -3463,15 +3467,16 @@ def start_scylla(node, _queue):
         self.run_func_parallel(func=stop_scylla, node_list=self.seed_nodes)  # pylint: disable=no-member
         # Then, update bin only on requested nodes
         self.run_func_parallel(func=update_scylla_bin, node_list=node_list)  # pylint: disable=no-member
-        # Start all seed nodes
-        self.run_func_parallel(func=start_scylla, node_list=self.seed_nodes)  # pylint: disable=no-member
-        # Start all non seed nodes
-        self.run_func_parallel(func=start_scylla, node_list=self.non_seed_nodes)  # pylint: disable=no-member
+        if start_scylla:
+            # Start all seed nodes
+            self.run_func_parallel(func=start_scylla, node_list=self.seed_nodes)  # pylint: disable=no-member
```

Comment:
is this start supposed to run before we upgrade at all?

Reply:
We don't need to start it before replacing, so I make it option, and disabled it.

The first start failed as expected because we disabled it in user format data.
The second start will be the start after node setup.

Reply:
my question was, if this code (that run start_scylla) happens before the code that upgrades the Scylla version?

also, in Asias test, IIUC what restarted it was systemd, and not SCT, so we need to stop scylla before that, or am I wrong?

Reply:
> my question was, if this code (that run start_scylla) happens before the code that upgrades the Scylla version?

Yes, it's before the upgrade.

> also, in Asias test, IIUC what restarted it was systemd, and not SCT, so we need to stop scylla before that, or am I wrong?

The start was caused by SCT.
```
2021-01-19T12:57:05+00:00  perf-latency-nemesis-new_sched-db-node-3da76a94-1 !NOTICE  | sudo:  centos : TTY=unknown ; PWD=/home/centos ; U
SER=root ; COMMAND=/bin/systemctl start scylla-server.service
2021-01-19T12:57:05+00:00  perf-latency-nemesis-new_sched-db-node-3da76a94-1 !INFO    | sudo: pam_unix(sudo:session): session opened for u
ser root by (uid=0)
2021-01-19T12:57:05+00:00  perf-latency-nemesis-new_sched-db-node-3da76a94-1 !INFO    | systemd: Starting Scylla Server...
....
2021-01-19T12:57:06+00:00  perf-latency-nemesis-new_sched-db-node-3da76a94-1 !INFO    | scylla: Scylla version 4.2.2-0.20201213.696ef24226 with build-id e82d9155c2a9f57c71b5de5575f2ae89e411863a starting ...

```